File: spring-grpc-client/app/src/main/java/com/example/springgrpc/app/GreetingController.java
package com.example.springgrpc.app;

import com.example.springgrpc.client.HelloGrpcClient;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

import java.util.Map;

@RestController
@RequestMapping("/api/greetings")
public class GreetingController {
    private final HelloGrpcClient helloGrpcClient;

    public GreetingController(HelloGrpcClient helloGrpcClient) {
        this.helloGrpcClient = helloGrpcClient;
    }

    @GetMapping
    public Map<String, String> greet(@RequestParam String name) {
        return Map.of("message", helloGrpcClient.sayHello(name).getMessage());
    }
}

File: spring-grpc-client-kotlin/app/src/main/kotlin/com/example/springgrpc/app/GreetingController.kt
package com.example.springgrpc.app

import com.example.springgrpc.client.HelloGrpcClient
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RequestParam
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/api/greetings")
class GreetingController(
    private val helloGrpcClient: HelloGrpcClient,
) {
    @GetMapping
    fun greet(@RequestParam name: String): Map<String, String> =
        mapOf("message" to helloGrpcClient.sayHello(name).message)
}
